pkg/server/api_plot.go

```diff
@@ -357,7 +357,9 @@ func makePlotsResponse(plotSeries map[string][]plot.Series, plotReq *PlotRequest
 					plotItem.Scale(plot.Value(scale))
 				}
 
-				seriesOptions[seriesItem.Name] = seriesItem.Options
+				if len(seriesItem.Options) != 0 {
```

Comment:
Hi there, sorry, the commit message wasn't that clear.
I am not sure, what the intended behaviour actually is. 
Probably the graph definition is broken. But I cannot judge which part is wrong and which behaviour is needed. Question is: where should the options (like unit, etc) reside: at group level or series level?

The behaviour changed between 0.2.0 and the latest git version:
0.2.0:
```json
    "groups": [
        {
            "name": "group0",
            "series": [
                {
                    "name": "load/load/longterm",
                    "metric": "load/load/longterm",
                    "options": null
                }
            ],
            "options": {
                "unit": "X"
            }
        },
        {
            "name": "load/load/midterm",
            "series": [
                {
                    "name": "load/load/midterm",
                    "metric": "load/load/midterm",
                    "options": {
                        "unit": "Y"
                    }
                }
            ],
            "options": {}
        }
```
latest:
```json
{
    "name": "Test Bug",
    ...
    "groups": [
        {
            "name": "group0",
            "series": [
                {
                    "name": "load/load/midterm/average",
                    "options": null
                }
            ],
            "options": {
                "unit": "Y"
            }
        },
        {
            "name": "load/load/longterm/average",
            "series": [
                {
                    "name": "load/load/longterm/average",
                    "options": null
                }
            ],
            "options": {
                "unit": "X"
            }
        }
    ],
}
```
as you can see, with the latest version, only groups have the Options set. Not series like previously.

With the current code, plot will first set the Options/Unit for the group and afterwarts overwrite it with the ones per series. It only matters for un-grouped metrics, as the names of the group and serie in the graph definition are the same in that case:
```go
for _, groupItem := range graph.Groups {
    seriesOptions[groupItem.Name] = groupItem.Options
    ...
    for _, seriesItem := range groupItem.Series {
    ...
        if len(seriesItem.Options) != 0 {
            seriesOptions[seriesItem.Name] = seriesItem.Options
       }
```
Additionally, with the last version, the graph definition looks different, when you first assign a unit and then grouping it afterwards. The UI is also inconsistent, here, since per series Options are not visible, as soon as a serie gets grouped. 
```json
{
    "name": "BUG 2",

    "groups": [
        {
            "name": "group0",
             "series": [
                {
                    "name": "load/load/midterm/average",
                     "options": {
                        "unit": "SER"
                    }
                }
            ],
            "options": {
                "unit": "GRP"
            }
        },
    ],
 }
```
so best would probably, to make clear if plot should provide per series options or per group options. Mixing them in plot doesn't seem to be a good option (?).

For me, the situation is quite hard to describe...
Does this explanation make any sense to you :-) ?